### PR TITLE
Tweak documentation for Sphinx

### PR DESCRIPTION
**Pull request by DougBurke**
This is a documentation-only change. There is no meaningful change to the
documentation; the changes are mostly sphinx-related (to avoid problems
generating HTML documentation, often because the suggested standards are
not 100% clear on what is supported), as well as a few minor clean ups
found while reviewing the documentation.

### CODE REVIEW

**Comment by DougBurke** [comment]
I claim the Travis failure here is not due to this PR. I'll update it once #338 is merged, but am asking for a review from @olaurino and @anetasie

File: sherpa/astro/ui/utils.py [hunk 1]
```diff
@@ -2338,10 +2346,10 @@ def set_staterror(self, id, val=None, fractional=False, bkg_id=None):
         val : array or scalar
            The systematic error.
         fractional : bool, optional
-           If ``False`` (the default value), then the ``val`` parameter is
-           the absolute value, otherwise the ``val`` parameter
+           If ``False`` (the default value), then the `val` parameter is
+           the absolute value, otherwise the `val` parameter
            represents the fractional error, so the absolute value is
-           calculated as ``get_dep() * val`` (and ``val`` must be
+           calculated as ``get_dep() * val`` (and `val` must be
```
Comment (olaurino) [inline sherpa/astro/ui/utils.py]: Shouldn't `get_dep() * val` be in single backquotes as interpreted text?

File: sherpa/astro/ui/utils.py [hunk 2]
```diff
@@ -2410,10 +2418,10 @@ def set_syserror(self, id, val=None, fractional=False, bkg_id=None):
         val : array or scalar
            The systematic error.
         fractional : bool, optional
-           If ``False`` (the default value), then the ``val`` parameter is
```
Comment (olaurino) [inline sherpa/astro/ui/utils.py]: Shouldn't `False`, `None`, etc. be in single backquotes like `load_arrays` etc?

Comment (DougBurke) [inline sherpa/astro/ui/utils.py]: The single and double tick marks have a different meaning: `foo` is a link/reference whereas ``foo`` is a "code fragment"; see http://www.sphinx-doc.org/en/stable/rest.html and links therein. We could actually have None/False/... as links, as they could link to the actual Python documentation for those items, but it's not clear whether it's worth it. So we want ``get_dep() * val`` to remain double quotes. 

The trouble with `foo` is that the documentation system can get confused and dump a warning about  an invalid link. I haven't really seen that with my latest attempt at the docs (#302), but I think I saw it in previous attempts.

There's also "tweaks" to the rules. For instance, the astropy (essentially the NumPy ones) docstring rules, which I'm trying to follow, say to use `foo` when referring to parameter values of a function [*], but it's not 100% clear if that is only in the parameter block or in the whole method docstring.

[*] see section 4 of http://docs.astropy.org/en/stable/development/docrules.html#sections

I admit that the current documentation is a mix. I am trying to move towards a consistent style, but haven't ironed out the edges yet.

Comment (DougBurke) [inline sherpa/astro/ui/utils.py]: For reference, the set_staterror call - when rendered in HTML - looks like http://hea-www.cfa.harvard.edu/~dburke/playground/sherpa/ui/api/sherpa.astro.ui.set_staterror.html

(I can't guarantee that this link will remain valid long term, but hopefully this PR will be merged before the link is changed).

**Comment by DougBurke** [comment]
Other examples are the two `unpack_data` calls (as they have slightly-different implementations):

http://hea-www.cfa.harvard.edu/~dburke/playground/sherpa/ui/api/sherpa.ui.unpack_data.html
http://hea-www.cfa.harvard.edu/~dburke/playground/sherpa/ui/api/sherpa.astro.ui.unpack_data.html

The first one is particularly improved by this PR (I unfortunately don't have a HTML rendition of the old version).

**Comment by olaurino** [comment]
Rebasing, original tip was 092c654.

**Comment by olaurino** [comment]
Again, should I merge this (assuming Travis' blessings)?

**Comment by DougBurke** [comment]
Unlike #337 these are more "mechanical" changes rather than substantive changes to the documentation, so I think these can be merged without a review by @anetasie

**Pull request merged**